### PR TITLE
Add Development Guidelines

### LINKED ISSUE #33: Add development guidelines
We need at least some guidelines related to installing `mcpp`, see #32.

### PR DESCRIPTION
**Pull request by mariaschett**
Fixes #33 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,32 +23,7 @@ installed. This helps with making the builds reproducible and the provenances
...
-local repository, by specifying `-git_root_dir`. In this case, the binary will
-be built from the repo, only if the latest commit matches the one specified in
-the config file.
-
-```bash
-$ bazel run  //cmd/build:main -- \
-  -config <path-to-transparent-release>/testdata/build.toml \
-  -git_root_dir <path-to-git-repo-root>
-```
+Check the [`development guidelines`](docs/development-guidelines.md) for a quick start to [`building and provenances`](docs/development-guidelines.md#building-and-provenances).
```
Comment (rbehjati) [inline README.md:26]: These commands are meant to be used by end-users not just the developers. So I am a bit unsure that it is a good idea to refer the users to the development guidelines for how to use the commands.

File: README.md [hunk 2]
```diff
@@ -60,23 +35,4 @@ and steps for building a binary from the listed materials. The verification
...
-local repository, by specifying `-git_root_dir`. In this case, the binary will
-be built from the repo, only if the latest commit matches the one specified in
-the config file.
-
-```bash
-$ bazel run  //cmd/verify:main -- \
-  -config <path-to-transparent-release>/schema/amber-slsa-buildtype/v1/example.json \
-  -git_root_dir <path-to-git-repo-root>
-```
+Check the [`development guidelines`](docs/development-guidelines.md) for a quick start to [`verifying provenances`](docs/development-guidelines.md#verifying-provenances).
```
Comment (rbehjati) [inline README.md:38]: Same as above :)

File: docs/development-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+You should see the following output on the console:
+
+```bash
+2022/04/14 09:08:17 The hash of the binary is: 15dc16c42a4ac9ed77f337a4a3065a63e444c29c18c8cf69d6a6b4ae678dca5c
+2022/04/14 09:08:17 Storing the provenance in <your-path>/transparent-release/provenance.json
+```
+
+To build from a local repository you can specify `-git_root_dir`. In this case, the binary will be built from the repo, only if the latest commit matches the one specified in the config file:
```
Comment (rbehjati) [inline docs/development-guidelines.md:29]: Maybe add that "otherwise the command will fail with an error".

File: README.md [hunk 4]
```diff
@@ -23,32 +23,7 @@ installed. This helps with making the builds reproducible and the provenances
 verifiable. The toml file should conform to the `BuildConfig` structure defined
 in the [`common`](/common/) package.
 
-The [`cmd/build`](/cmd/build/) command line tool described above can be used
```
Comment (rbehjati) [inline README.md:26]: I think this paragraph is important and should be included. This describes how to provide a config file similar to `testdata/build.toml`.

File: README.md [hunk 5]
```diff
@@ -23,32 +23,7 @@ installed. This helps with making the builds reproducible and the provenances
...
-for building and releasing the binaries, and at the same time for generating a
-corresponding provenance file. To use this tool, the developers need to provide
```
Comment (rbehjati) [inline README.md:28]: "the developers" here refers to the developers of the project who want to build and release their libraries/binaries using transparent-release, not the transparent-release development team :)

**Review by rbehjati — APPROVED** [review]
Very nice. Thanks :)

There is one paragraph that I think should be added back to the README. 
The line breaks are not quite right. I think we need to add `prettier` to the formatting script.

File: README.md (hunk 4 above)
Comment (mariaschett) [reply README.md:26]: Added it back in---sorry!

File: README.md (hunk 1 above)
Comment (mariaschett) [reply README.md]: Moved back to README and added a link to the development guidelines for building.

File: README.md (hunk 4 above)
Comment (rbehjati) [reply README.md:26]: Thanks :)

File: docs/development-guidelines.md (hunk 3 above)
Comment (mariaschett) [reply docs/development-guidelines.md]: Done.

**Review by mariaschett — COMMENTED** [review]
Done, and added prettierrc.yaml. Let's see if that works.

**Pull request merged**